Mathlib/Algebra/Category/AlgebraCat/Monoidal.lean

```diff
@@ -0,0 +1,111 @@
+/-
+Copyright (c) 2023 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+-/
+import Mathlib.CategoryTheory.Monoidal.Braided
+import Mathlib.CategoryTheory.Monoidal.Transport
+import Mathlib.Algebra.Category.AlgebraCat.Basic
+import Mathlib.Algebra.Category.ModuleCat.Monoidal.Basic
+import Mathlib.RingTheory.TensorProduct
+
+/-!
+# The monoidal category structure on R-algebras
+-/
+
+open CategoryTheory
+
+universe v u
+
+variable {R : Type u} [CommRing R]
+
+namespace AlgebraCat
+
+noncomputable section
+
+namespace instMonoidalCategory
+
+open scoped TensorProduct
+
+/-- Auxiliary definition used to fight a timeout when building
+`AlgebraCat.instMonoidalCategory`. -/
+@[simps!]
+noncomputable abbrev tensorObj (X Y : AlgebraCat.{u} R) : AlgebraCat.{u} R :=
+  of R (X ⊗[R] Y)
+
+/-- Auxiliary definition used to fight a timeout when building
+`AlgebraCat.instMonoidalCategory`. -/
+noncomputable abbrev tensorHom {W X Y Z : AlgebraCat.{u} R} (f : W ⟶ X) (g : Y ⟶ Z) :
+    tensorObj W Y ⟶ tensorObj X Z :=
+  Algebra.TensorProduct.map f g
+
+/-- Auxiliary definition used to fight a timeout when building
+`AlgebraCat.instMonoidalCategory`. -/
+@[simps!]
+abbrev tensorUnit : AlgebraCat.{u} R := of R R
+
+/-- Auxiliary definition used to fight a timeout when building
+`AlgebraCat.instMonoidalCategory`. -/
+noncomputable abbrev associator (X Y Z : AlgebraCat.{u} R) :
+    tensorObj (tensorObj X Y) Z ≅ tensorObj X (tensorObj Y Z) :=
+  (Algebra.TensorProduct.assoc R X Y Z).toAlgebraIso
+
+open MonoidalCategory
+
+theorem forget₂_map_associator_hom (X Y Z : AlgebraCat.{u} R) :
+    (forget₂ (AlgebraCat R) (ModuleCat R)).map (associator X Y Z).hom
+      = (α_
+        (forget₂ _ (ModuleCat R) |>.obj X)
+        (forget₂ _ (ModuleCat R) |>.obj Y)
+        (forget₂ _ (ModuleCat R) |>.obj Z)).hom := by
+  rfl
+
+set_option maxHeartbeats 400000 in
+theorem forget₂_map_associator_inv (X Y Z : AlgebraCat.{u} R) :
+    (forget₂ (AlgebraCat R) (ModuleCat R)).map (associator X Y Z).inv
+      = (α_
+        (forget₂ _ (ModuleCat R) |>.obj X)
+        (forget₂ _ (ModuleCat R) |>.obj Y)
+        (forget₂ _ (ModuleCat R) |>.obj Z)).inv := by
+  rfl
+
+end instMonoidalCategory
+
+open instMonoidalCategory
+
+set_option maxHeartbeats 800000 in
+noncomputable instance instMonoidalCategory : MonoidalCategory (AlgebraCat.{u} R) :=
+  Monoidal.induced
+    (forget₂ (AlgebraCat R) (ModuleCat R))
+    { tensorObj := instMonoidalCategory.tensorObj
+      μIsoSymm := fun X Y => eqToIso rfl
```

Comment:
```suggestion
      μIsoSymm := fun X Y => Iso.refl _
```
(And similarly for ` εIsoSymm`.)
I understand that the definitional properties of these fields do not matter for the definition of the monoidal category structure, but it matters for the compatibilities of the forgetful functor with respect to the tensor product.

Reply:
I believe these are definitionally equal anyway, but either your version broke the Aesop proofs, or it timed out. Perhaps the problem I'm remembering no longer exists

Reply:
Yes, indeed, there are defeq, but when unfolding the definitions, `Iso.refl _` would presumably behave better than `eqToIso rfl`. Yesterday, this file was compiling on my computer with this change.

Reply:
I agree, this seems to work. It was https://github.com/leanprover-community/mathlib4/pull/7244 where I ran into trouble, but perhaps that went away too after the latest merge.

I've made the change you suggest, thanks for asking me to revisit it. I also timed it with `count_heartbeats`; your version is faster!